src/transformers/utils/deprecation.py

```diff
@@ -0,0 +1,148 @@
+        new_name (`Optional[str]`, *optional*):
+            The new name for the deprecated keyword argument. If specified, the deprecated keyword argument will be replaced with this new name.
+        warn_if_greater_or_equal_version (`bool`, *optional*, defaults to `False`):
+            Whether to show warning if current `transformers` version is greater or equal to the deprecated version.
+        raise_if_greater_or_equal_version (`bool`, *optional*, defaults to `False`):
+            Whether to raise `ValueError` if current `transformers` version is greater or equal to the deprecated version.
+        raise_if_both_names (`bool`, *optional*, defaults to `False`):
+            Whether to raise `ValueError` if both deprecated and new keyword arguments are set.
+        additional_message (`Optional[str]`, *optional*):
+            An additional message to append to the default deprecation message.
+
+    Raises:
+        ValueError:
+            If raise_if_greater_or_equal_version is True and the current version is greater than or equal to the deprecated version, or if raise_if_both_names is True and both old and new keyword arguments are provided.
+
+    Returns:
+        Callable:
+            A wrapped function that handles the deprecated keyword arguments according to the specified parameters.
+
+    Example usage with renaming argument:
+
+        ```python
+        @deprecate_kwarg("reduce_labels", new_name="do_reduce_labels", version="6.0.0")
+        def my_function(do_reduce_labels):
+            print(do_reduce_labels)
+
+        my_function(reduce_labels=True)  # Will show a deprecation warning and use do_reduce_labels=True
+        ```
+
+    Example usage without renaming argument:
+
+        ```python
+        @deprecate_kwarg("max_size", version="6.0.0")
+        def my_function(max_size):
+            print(max_size)
+
+        my_function(max_size=1333)  # Will show a deprecation warning
+        ```
+
+    """
+
+    deprecated_version = packaging.version.parse(version)
+    current_version = packaging.version.parse(__version__)
+    is_already_deprecated = current_version >= deprecated_version
+
+    if is_already_deprecated:
+        version_message = f"and removed starting from version {version}"
+    else:
+        version_message = f"and will be removed in version {version}"
+
+    def wrapper(func):
+        @wraps(func)
+        def wrapped_func(*args, **kwargs):
+            minimum_action = None
+            message = None
+
+            # deprecated kwarg and its new version are set for function call -> replace it with new name
+            if old_name in kwargs and new_name in kwargs:
+                minimum_action = Action.RAISE if raise_if_both_names else Action.NOTIFY
+                message = f"Both `{old_name}` and `{new_name}` are set. Using `{new_name}={kwargs[new_name]}` and ignoring deprecated `{old_name}={kwargs[old_name]}`."
+                kwargs.pop(old_name)
```

Comment:
I was wondering about this specific use-case, I may be missing something: if I do

```python
proc = BeitImageProcessor(reduce_labels=True, weird_arg=True)
# UserWarning: The following named arguments are not valid for `BeitImageProcessor.__init__` and were ignored: 'weird_arg'
```

I'm getting a warning here, but if I call the two versions of the argument (which can be strings it seems, but that's unrelated)

```python
proc = BeitImageProcessor(reduce_labels="any value really", do_reduce_labels="another value")
print( proc(images=np_img)['pixel_values'][0].shape )
# (3, 224, 224)
# no warning here
```
but it seems it should be the same log level?

Reply:
Oh, this is a nice catch! Will fix and add a test!

Reply:
addressed in https://github.com/huggingface/transformers/pull/30799/commits/9156618dc3b0aaf1e68460aa6f7e43f83e781700